pandera/backends/pyspark/decorators.py

```diff
@@ -126,3 +131,54 @@ def _get_check_obj():
         return wrapper
 
     return _wrapper
+
+
+def cache_check_obj():
+    """This decorator evaluates if `check_obj` can be cached before validation.
```

Comment:
Noticing, this isn't true for the other dectorators in this file either but would it make sense to cleariy in the docstring that this is a decorator factory and should decorated with `cache_check_obj`?

Reply:
Something similar to

```python
    """
    A decorator factory that creates a decorator to evaluate if `check_obj` can be cached before validation.

    As each new data check added to the Pandera schema by the user triggers 
    a new Spark action, Spark reprocesses the `check_obj` DataFrame multiple times.
    To prevent this waste of processing resources and to reduce validation times 
    in complex scenarios, the decorator created by this factory caches the 
    `check_obj` DataFrame before validation and unpersists it afterwards.

    The behavior of the resulting decorator depends on the `PANDERA_PYSPARK_CACHING` 
    environment variable.

    Usage:
        @cache_check_obj()
        def your_function(...):
            ...

    Note: This is not a direct decorator but a factory that returns a decorator.
    """
```

Reply:
Indeed!